app/retail/views.py

```diff
@@ -994,6 +994,27 @@ def activity(request):
 
     return TemplateResponse(request, 'activity.html', context)
 
+@ratelimit(key='ip', rate='30/m', method=ratelimit.UNSAFE, block=True)
+def create_status_update(request):
+    response = {}
+    if request.POST:
+        profile = request.user.profile
+        kwargs = {
+            'activity_type': 'status_update',
+            'metadata': {
+                'title': request.POST.get('data')
+            }
+        }
+        kwargs['profile'] = profile
```

Comment:
You can combine this line with the lines above it to create the `kwargs` dictionary in one statement